_posts/2018-02-28-Update-to-Java11.md

```diff
@@ -0,0 +1,124 @@
+
+Java 10 is to be released any moment now. Java 9 public updates are also destined to end soon, so you should update to Java 10 soon if you like those updates. Java 10 is going to be another short term support update, with Java 11 being a long term support update, like Java 8.
+
+![Some of the JDK 10/11 goodies](/img/java11/jshell_inference.png)
+
+Java 10 will have type inference for local variable type (YAY!), so your code will look like this:
+
+```Java
+// Upcoming Java 10 code
+var productList = new ArrayList<String>();
+productList.add("Macbook Pro");
+var productStream = productList.stream();
+productStream.forEach(System.out::println);
+```
+
+Note a few things. This requires you to define the generic typing on the right side, so it can be inferred. No diamond operator for you. Secondly, there's no 'val', only var. Meaning, it's just a normal mutable variable declaration, with a bit syntactic sugar. Still nice. You can test it immediately with Java 10/11 early access build, and soon with Java 10 GA build.
+
+It will not work for method/constructor parameters or return types, member variables/fields, or catch formals. Only for logic within methods, including loop indexes.
+
+Of course, we still have all Java 9 goodies such as new Stream functions:
+
+```
+jshell> IntStream.range(0,10).takeWhile(x -> x < 5).forEach(System.out::println)
+0
+1
+2
+3
+4
+```
+
+If you update to any of these Java versions, 9, 10, 11, you get access to JShell and goodies like that. JShell that was introduced in Java 9, starts up now significantly faster, making it much better REPL experience for those who value it.
+
+There is some Docker love coming up in Java 10 - improved container detection and resource configuration usage. So Java becomes more aware of its environment, especially running in containers.
+
+As always, better performance, improved garbage collection, experimental Jit compiler, open sourced root certificates, yada yada yada. Expect the major releases to happen much faster, even twice a year. Long term support releases will come every 3 years - but to get that three years support updates you will probably need to pay for it, too. See the blog link at end of article on free, secure, stable.
+
+JDK 11 will continue the modularity work and more aggressively remove modules from core Java, making it lighter and easier to update. Next on list are CORBA and Java EE modules. You can still get them, on top of core Java, when you need them. But they're not bundled with every micro-service, freeing up 22 megabytes and 9 modules from it. It goes leaner and meaner.
+
+Project Panama and Valhalla are still coming up, value types, generic specialization, native function calling from JVM, etc. Amber project started with local variable type inference, but enums will be having much more love in future versions, too. At this early stage, not much concrete to show yet.
```

Comment:
"coming up, with…"

"Amber project" -> "Project Amber"?

"enums will be ~having~ getting"

"nothing concrete" might be better.